### ISSUE TITLE
Optional String Field with Options leads to: "Expected Iterable, but did not find one for field"

### ISSUE THREAD

**Issue by gossi** [tinacms/tinacms#4399]
### Describe the issue that you're seeing. Any Loom videos or screenshots usually help a lot!

I schemed my existing markdown files for usage in `tina`, a collection named `exercise`. The frontmatter consists of two fields `locomotion` and `personal`, both are lists of strings with given options and both fields are **optional** !!

Here is how I wrote the schema for them:

```ts
const schema = defineSchema({
  collections: [
    {
      name: 'exercises',
      label: 'Exercises',
      path: 'content/exercises',
      fields: [
        {
          type: "string",
          name: "title",
          label: "Title",
          isTitle: true,
          required: true
        },
        {
          label: 'Locomotion',
          name: 'locomotion',
          type: 'string',
          list: true,
          options: [
            {
              value: 'pedes',
              label: 'Pedes',
            },
            {
              value: 'unicycle',
              label: 'Unicycle'
            }
          ]
        }
      ]
      // ... same for `personal` + more fields
    }
  ]
});
```

Now I queried them following the [data fetching guides](https://tina.io/docs/features/data-fetching/). 

1) Query the individual exercise: 

```ts
const ex = await client.queries.exercises({ relativePath: `${id}.md` });
```

✅ That works, regardless whether `personal` or `locomotion` are given.

2) Query the collection:

```ts
const exercisesResponse = await client.queries.exercisesConnection(); // <-- this line yields the error

const exercises = exercisesResponse.data.exercisesConnection.edges?.map((ex) => {
  return { slug: ex?.node?._sys.filename };
});
```

❌ That failed with error message below:

```
GraphQL request:41:3
40 |   title
41 |   locomotion
   |   ^
42 |   personal
Expected Iterable, but did not find one for field "Exercises.locomotion".
```

or since not all exercises do need these fields, there are a bunch of these errors:

<img width="641" alt="Bildschirmfoto 2023-12-16 um 14 22 58" src="https://github.com/tinacms/tinacms/assets/283700/74ba4648-c52d-4b19-bcc6-463a01222500">

I grasped as much documentation as possible to help myself, but this is how far I got. I had the "find all" query at first and then sorta prooved my own config by writing the individual query as second to see if that might fail, too.

In the individual query the fields are `null` which would be my expected result for the connection query, too.

### Reproduction

https://github.com/gossi/unidancing/pull/27

### System Info

```shell
System:
    OS: macOS 14.1
    CPU: (11) x64 Apple M3 Pro
    Memory: 133.70 MB / 36.00 GB
    Shell: 3.6.4 - /opt/homebrew/bin/fish
  Binaries:
    Node: 18.16.0 - ~/.volta/tools/image/node/18.16.0/bin/node
    Yarn: 1.22.19 - ~/.volta/tools/image/yarn/1.22.19/bin/yarn
    npm: 9.5.1 - ~/.volta/tools/image/node/18.16.0/bin/npm
    Watchman: 2023.12.04.00 - /opt/homebrew/bin/watchman
  Browsers:
    Brave Browser: 103.1.41.100
    Chrome: 120.0.6099.109
    Edge: 120.0.2210.77
    Safari: 17.1
  npmPackages:
    @tinacms/cli: ^1.5.39 => 1.5.39
    tinacms: ^1.5.28 => 1.5.28
```
```


### Validations

- [X] Follow our [Code of Conduct](https://github.com/tinacms/tinacms/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://tina.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/tinacms/tinacms/issues) that reports the same bug.

**Comment by kldavis4** [comment]
I suspect it is that some of your content is not actually a list - see this example https://github.com/gossi/unidancing/blob/ssr/web/content/exercises/copycat.md?plain=1#L3

**Comment by gossi** [comment]
Thank you, that's it. My custom code was in the type `string | string[]`.

I had a false positive verification though. In the tina admin, I checked the admin UI panel for that sort of combination, and the UI looked happy about it, I could use it.

Until I saved it, which then turned into:

```yaml
locomotion:
 - p
 - e
 - d
 - e
 - s
```

My optimistic expectation was some sort of validation beforehand, either in the Admin UI a message about that file, or in CI when this it is started, that it scans the markdown files and reports on errors where it doesn't match the schema?
